src/Microsoft.AspNet.Mvc.Core/Formatters/StringOutputFormatter.cs

```diff
@@ -31,13 +31,9 @@ public override bool CanWriteResult(OutputFormatterCanWriteContext context)
 
             // Ignore the passed in content type, if the object is string
             // always return it as a text/plain format.
-            if (context.ObjectType == typeof(string))
-            {
-                return true;
-            }
-
-            if (context.Object is string)
+            if (context.ObjectType == typeof(string) || context.Object is string)
```

Comment:
Yeah, that's true that it shouldn't be needed. Being cavalier about removing things is what broke these in the first place. Let's sync up on removing `context.Object` tomorrow if possible and revisit this as a forcing function.